state/snapshotStatistics.go

```diff
@@ -52,7 +52,7 @@ func (ss *snapshotStatistics) AddTrieStats(trieStats *statistics.TrieStatsDTO) {
 	ss.mutex.Lock()
 	defer ss.mutex.Unlock()
 
-	ss.trieStatisticsCollector.Add(trieStats)
+	ss.trieStatisticsCollector.Add(*trieStats)
```

Comment:
this can easily panic if I send the trie stats as nil. We need to change here also

Reply:
pushed, kept using pointer but added check if nil